### DISCUSSION TITLE
Preventing tons of empty projects from clogging up my stuff

### DISCUSSION THREAD

**Discussion by janSipiki** [ScratchAddons/ScratchAddons#4476] [Ideas]
I'm not sure about you, but my "My Stuff" page is filled to the brim with a ton of blank projects;
These things were made from me accidentally clicking the create button on the home screen. Since these projects save **immediately** after their creation, they end up in My Stuff, where they clog up the page and make it hard to find actual things I've put effort into. Adding an addon that only save after a block is placed or a costume is edited, and/or having a way to bulk delete unshared, empty projects would really help.

P.S. I know I could just delete them manually or not be a clumsy idiot but I'm lazy and it would eventually get just as bad as before..

![Screenshot (87)](https://user-images.githubusercontent.com/82186382/162334051-ebb2b8fa-a457-433c-a0b6-d738be74d642.png)

**Comment by apple502j** [comment]
You can actually remove the Create button using custom navigation bar addon, fyi 😀

**Reply by Secret-chest** [reply]
> You can actually remove the Create button using custom navigation bar addon, fyi grinning

I'd like to come up with an actual solution: make an addon to disable auto-save until you click the save button once.

**Reply by Samq64** [reply]
Doesn't that already exist?

**Reply by Norbiros** [reply]
It disables it always. I think in this new feature we should disable autosave, until button is first time pressed, after that - autosave works normally...

**Reply by Cool-Scratcher** [reply]
> It disables it always. I think in this new feature we should disable autosave, until button is first time pressed, after that - autosave works normally...

And to prevent accidentally forgetting, there should be a warning because forgetting to click the save button once is something that sounds like I might do it...

**Reply by Norbiros** [reply]
Maybe this alert: "You have unsaved changes. Leave/Stay". 

Also we can check if projects name starts with " Untitled ", if no - autosave works normally. 

But thinking in other way: maybe " delete all" button (you can easily select projects to delete) ?  *This feature can be used in not just cleaning up untitled projects, if we chose other idea to fix this problem, we can still create this add-on."

**Reply by DatGamerYoloYT** [reply]
what if your making something that starts with untitled?
(Example: Untitled Platformer Game)

**Reply by Norbiros** [reply]
You can use regex. Check for project that starte with "Untitled Project " and check IF element after it is intenger, and after it you don't have anything else?

**Comment by Chiroyce1** [comment]
Yeah this happens to me as well lol 😛 @apple502j maybe an addon that introduces the delete button on the editor/project page itself? thats a good idea.. i still use the create button in the navbar so removing it isn't a solution...

**Reply by Norbiros** [reply]
I think that secret-chest's idea is better. When i create new project, I test some functionality of SA and close tab. This is nice idea, but it doesn't fix this problem IMAO

**Comment by rimopa** [comment]
This happens to me too, but I think the best solution is not add a project in My stuff until it is modified (someone already suggested this).

**Comment by WorldLanguages** [comment]
An addon that deletes projects in behalf of the user would probably be rejected.
Maybe the `Confirmations` addon could have a setting that lets the user confirm they want to create a new project after clicking the "Create" button?

**Reply by Norbiros** [reply]
Okay. We can add something like:

"Do you want create new project?"
Yes (create new project)
No, edit last one (when you click it it moves you to editor of latest project)
Cancel (don't do anything)

**Reply by WorldLanguages** [reply]
> No, edit last one (when you click it it moves you to editor of latest project)

I'm not sure about this one. Just go to my stuff and click on the first project.

**Reply by Norbiros** [reply]
Yes. But you know, waiting for opening my stuff (ot may także long) and selecting gile, moving your mouse, clicking, to much work for me lol. I am using that now, becouse i changed create link to my stuff... But if script for that may be difficult, or it gonna take a same amount of time as doing it manually okay.

Also, I think this setting should be off by default maybe?

**Comment by CST1229** [comment]
Or maybe an (internally) super complex addon that overrides POSTs to projects.scratch.mit.edu (only the root) and then creates the project during the first non-creation save, and probably does other things for things like api.scratch.mit.edu requests.

**Comment by Mbrick2** [comment]
Maybe `1 Time Projects`
